### PR TITLE
Spring2020 nexus

### PR DESCRIPTION
**Pull request by phyy-nx**
NeXus related commits for next release

### CODE REVIEW

File: format/nexus.py [hunk 1]
```diff
@@ -1085,74 +1110,49 @@ def find_scanning_axis(obj):
             return o
 
 
-class ScanFactory(object):
```
Comment (Anthchirp) [inline format/nexus.py:1088]: This pseudoclass was turned into a function. Fully support this, however in the context of a stable release this is an API change on a public name. I will put the class back in in a separate commit.

File: format/nexus.py [hunk 2]
```diff
@@ -535,26 +538,41 @@ class BeamFactory(object):
     A class to create a beam model from NXmx stuff
     """
 
-    def __init__(self, obj, index=None):
+    def __init__(self, obj):
```
Comment (Anthchirp) [inline format/nexus.py:541]: This changes the constructor signature and class behaviour on a public class. Can the fixes be backported as a minimal changeset that retain the current class behaviour? Alternatively I suggest a second class is introduced with the changed behaviour.

File: format/nexus.py (hunk 1 above)
Comment (Anthchirp) [reply format/nexus.py:1088]: done

**Comment by Anthchirp** [comment]
Added my suggested changes. @phyy-nx could you have a look? If you're happy with those all I'd need for the release is a bullet point summary of what your changes mean in practice for a user (for the [release page](https://github.com/dials/dials/releases/tag/v2.2.0)).

eg. for the GW/BW commit you picked this will be

- Fix reading of single-image Eiger files (#164)

**Comment by phyy-nx** [comment]
Hi, so dials 2.2 is pretty divergent from master then, right?  Will 2.2.5 be branched cleanly from master or is the 2.2 branch it's own thing until 2.2 support stops?

Regardless, these changes are fine.  Here's the summary:

- Minimal fixes needed to get HDF5 files written in py3 h5py to be read
- Fix reading of single-image Eiger files (#164)
- Add support for multi-wavelength NeXus files
- Update NeXus readers to find beam in both sample and instrument groups
- NeXus: better handle absence of gonio/scan

**Comment by ndevenish** [comment]
Yes, for us, all releases diverge from master when a minor (major.minor.patch) release is made - this means we can control precisely what goes into patch releases, and master development can go onwards without accidentally breaking something in release.

There's a new document at: https://github.com/dials/dials/wiki/Making-an-X.Y-DIALS-Release#repository-tags-and-branch-structure that explains the repository structure we use for releases:
```
        (towncrier for 3.0)
            ┊
o───M───o───T───o───o───o───o  master (v3.dev-6)
    ┊        ╲
 tag:v3.dev   ╲   o───o───o───o───o───o───o  dials-next
               ╲ ╱         ╲           ╲
                R───────────P───────────P    dials-3.0
                ┊           ┊           ┊
         tag:v3.0.0      (3.0.1)     (3.0.2)
```
It's examples are for v3.0 but we've used the branch structure and version labelling scheme since [DIALS 1.2](https://github.com/dials/dials/tree/dials-1.2), and the 2.2 LTS stuff just uses a separate `dials-next-2.2` branch.

**Comment by Anthchirp** [comment]
Squashed into linked commit above in dials-next-2.2 branch. Will go into next release.

**Pull request closed without merging**